lyrics-core/src/main/java/com/flipkart/lyrics/interfaces/AnnotationSpec.java

```diff
@@ -0,0 +1,56 @@
+package com.flipkart.lyrics.interfaces;
+
+import com.flipkart.lyrics.Song;
+import com.flipkart.lyrics.interfaces.model.NameFormatArgs;
+import com.flipkart.lyrics.interfaces.typenames.ClassName;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * @author kushal.sharma on 10/08/17.
+ */
+public class AnnotationSpec {
+    public final Class<?> clazz;
+    public final ClassName className;
+    public final List<NameFormatArgs> members = new ArrayList<>();
+
+    public AnnotationSpec(Builder builder) {
+        this.clazz = builder.clazz;
+        this.className = builder.className;
+        this.members.addAll(builder.members);
+    }
+
+    public static Builder builder(Class<?> clazz) {
+        return Song.factory.createAnnotationBuilder(clazz);
+    }
+
+    public static Builder builder(ClassName className) {
+        return Song.factory.createAnnotationBuilder(className);
+    }
+
+    public Object getAnnotationSpec() {
+        return null;
+    }
+
+    public static abstract class Builder {
+        private Class<?> clazz;
+        private ClassName className;
+        private final List<NameFormatArgs> members = new ArrayList<>();
+
+        protected Builder(Class<?> clazz) {
+            this.clazz = clazz;
```

Comment:
lets restrict to just one constructor as we can always create a ClassName out of a class